src/operation.js

```diff
@@ -201,21 +235,23 @@ export class Operation {
       return false;
     }
 
-    switch (true) {
+    if (
       // == 0
-      case !allowZero && amount.isZero():
+      (!allowZero && amount.isZero()) ||
       // < 0
-      case amount.isNegative():
+      amount.isNegative() ||
       // > Max value
-      case amount.times(ONE).greaterThan(new BigNumber(MAX_INT64).toString()):
-       // Decimal places (max 7)
-      case amount.decimalPlaces() > 7:
+      amount.times(ONE).greaterThan(new BigNumber(MAX_INT64).toString()) ||
+      // Decimal places (max 7)
+      amount.decimalPlaces() > 7 ||
       // NaN or Infinity
-      case (amount.isNaN() || !amount.isFinite()):
-        return false;
-      default:
-       return true;
+      amount.isNaN() ||
+      !amount.isFinite()
+    ) {
```

Comment:
This was part of #131 to improve [Cognitive Complexity](https://docs.codeclimate.com/docs/cognitive-complexity). Is linter feeling bad about this code?

Reply:
Interesting. I see what that article is trying to say, but I think the switch statement is too clever: switch statements aren't usually used like this, so it could easily be misread. I generally favor clear (in this case, recognizable) code over clever code. (Also note that "switch statements" is on the same list of breaks in flow as "sequences of logical operators".)